src/app/social-media/social-media-input/social-media-input.component.ts

```diff
@@ -0,0 +1,142 @@
+import { Component, OnInit, AfterViewInit, OnDestroy } from '@angular/core';
+import { Router, ActivatedRoute } from '@angular/router';
+import { SocialMediaType } from '../../view-models/social-media-type';
+import { SocialMediaPostExtended } from '../../view-models/social-media-post-extended';
+import { NavmenuService } from '../../services/navmenu.service';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { DeletePostConfirmationModalComponent } from '../delete-post-confirmation-modal/delete-post-confirmation-modal.component';
+import { AddSocialMediaPostModalComponent } from '../add-social-media-post-modal/add-social-media-post-modal.component';
+import { SocialMediaPostsService } from '../../services/socialMediaPosts.service';
+import { SocialMediaRenderService } from '../../services/socialMediaRender.service';
+
+@Component({
+  selector: 'app-social-media-input',
+  templateUrl: './social-media-input.component.html',
+  styleUrls: ['./social-media-input.component.scss']
+})
+
+export class SocialMediaInputComponent implements OnInit, AfterViewInit, OnDestroy {
+  socialmedia: SocialMediaPostExtended[];
+  selectedSocialMedia: SocialMediaPostExtended[];
+
+  socialmediatypes: SocialMediaType[];
+  filterBySocialMediaType: string;
+
+  constructor(
+    private router: Router,
+    private activatedRoute: ActivatedRoute,
+    public nav: NavmenuService,
+    private modal: NgbModal,
+    private socialMediaService: SocialMediaPostsService,
+    private socialMediaRenderService: SocialMediaRenderService) {
+    this.router.routeReuseStrategy.shouldReuseRoute = () => false;
+  }
+
+  ngOnInit() {
+    this.nav.hide();
+    this.activatedRoute.data.subscribe(data => {
+      this.socialmedia = data['socialmedia'];
+    });
+    const selectedSocialmediatypes = [];
+
+    this.socialmedia.forEach(post => {
+      if (selectedSocialmediatypes.indexOf(post.mediaType) === -1) {
+        selectedSocialmediatypes.push(post.mediaType);
+        this.socialMediaRenderService.loadWidgets(post.mediaType);
+      }
+    });
+  }
+
+  ngAfterViewInit() {
+    const selectedSocialmediatypes = [];
+    if (this.socialmedia !== null || this.socialmedia.length > 0) {
+      this.socialmedia.forEach(post => {
+        if (selectedSocialmediatypes.indexOf(post.mediaType) === -1) {
+          selectedSocialmediatypes.push(post.mediaType);
+          this.socialMediaRenderService.loadWidgets(post.mediaType);
+        }
+      });
+    }
+  }
+
+  ngOnDestroy() {
+    this.nav.show();
+  }
+
+  close() {
+    this.router.onSameUrlNavigation = 'reload';
+    this.router.navigate(['social-media-input']);
+  }
+
+  backToSocialMediaList() {
+    this.router.onSameUrlNavigation = 'reload';
+    this.router.navigate(['social-media-list']);
+  }
+
+  pinSocialMediaPost(post: SocialMediaPostExtended) {
+    const updatedPost = post;
+    if (updatedPost.sortOrder === 0) {
+      updatedPost.sortOrder = 1;
```

Comment:
updatedPost is a const so you shouldn't change it. Change updatedPost to a let if you want to make these changes like this. I would prefer to see this happen in the response of the API call. A nice trick for passing in most of an object with some slight changes is the spread operator. I'll give an example of the api call:

``this.socialMediaService.updateSocialMediaPost(updatedPost.id, { sortOrder: newSortOrder, ...updatedPost })``

This will grab any data from updatedPost that isn't already included in the object (ie sortOrder)